packages/snaps-execution-environments/src/common/endowments/console.ts

```diff
@@ -0,0 +1,74 @@
+import { rootRealmGlobal } from '../globalObject';
+import { EndowmentFactoryOptions } from './commonEndowmentFactory';
+
+export const consoleAttenuatedMethods = new Set([
+  'log',
+  'assert',
+  'error',
+  'debug',
+  'info',
+  'warn',
+]);
+
+/**
+ * Create a a {@link console} object, with the same properties as the global
+ * {@link console} object, but with some methods replaced.
+ *
+ * @param options - Factory options used in construction of the endowment.
+ * @param options.snapId - The id of the snap that will interact with the endowment.
+ * @returns The {@link console} object with the replaced methods.
+ */
+function createConsole({ snapId }: EndowmentFactoryOptions = {}) {
+  const keys = Object.getOwnPropertyNames(
+    rootRealmGlobal.console,
+  ) as (keyof typeof console)[];
+
+  const attenuatedConsole = keys.reduce((target, key) => {
+    if (consoleAttenuatedMethods.has(key)) {
+      return target;
+    }
+
+    return { ...target, [key]: rootRealmGlobal.console[key] };
+  }, {});
+
+  return harden({
+    console: {
+      ...attenuatedConsole,
+      log: (message?: any, ...optionalParams: any[]) => {
+        rootRealmGlobal.console.log(`[SNAP LOG | snapId: ${snapId}]`);
+        rootRealmGlobal.console.log(message, ...optionalParams);
+      },
```

Comment:
Rather than logging an extra message, we could also just prepend the message.
```suggestion
      log: (message?: any, ...optionalParams: any[]) => {
        rootRealmGlobal.console.log(`[Snap: ${snapId}] ${message}`, ...optionalParams);
      },
```

Reply:
Yeah, I'll do this.

Reply:
That will break when using substitutions: The message must be the first parameter. Actually, something like this may be better:

```ts
function getMessage(snapId: SnapId, message: unknown, ...args: unknown[]) {
  const prefix = `[Snap: ${snapId}]]`;

  // If the first argument is a string, prepend the prefix to the message, and keep the
  // rest of the arguments as-is.
  if (typeof message === 'string') {
    return [`${prefix} ${message}, ...args];
  }

  // Otherwise, the `message` is an object, array, etc., so add the prefix as a separate
  // message to the arguments.
  return [prefix, message, ...args];
}

console.log(...getMessage(...));
```